src/autoscaler/metricsforwarder/forwarder/syslog_emitter_test.go

```diff
@@ -0,0 +1,121 @@
+package forwarder_test
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"code.cloudfoundry.org/app-autoscaler/src/autoscaler/metricsforwarder/config"
+	"code.cloudfoundry.org/app-autoscaler/src/autoscaler/metricsforwarder/forwarder"
+	"code.cloudfoundry.org/app-autoscaler/src/autoscaler/models"
+	"code.cloudfoundry.org/lager/v3"
+	"code.cloudfoundry.org/loggregator-agent-release/src/pkg/egress/syslog"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("SyslogEmitter", func() {
+	var (
+		listener net.Listener
+		err      error
+		port     int
+		conf     *config.Config
+		tlsCerts models.TLSCerts
+		emitter  forwarder.MetricForwarder
+	)
+
+	BeforeEach(func() {
+		port = 10000 + GinkgoParallelProcess()
+		listener, err = net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", port))
+		Expect(err).ToNot(HaveOccurred())
+		tlsCerts = models.TLSCerts{}
+		Expect(err).ToNot(HaveOccurred())
+
+	})
+
+	JustBeforeEach(func() {
+		host, port, err := net.SplitHostPort(listener.Addr().String())
+		Expect(err).ToNot(HaveOccurred())
+		fmt.Println(host, port)
+
+		portNumber, err := strconv.Atoi(port)
+		Expect(err).ToNot(HaveOccurred())
+
+		conf = &config.Config{
+			SyslogConfig: config.SyslogConfig{
+				ServerAddress: host,
+				Port:          portNumber,
+				TLS:           tlsCerts,
+			},
+		}
+
+		//TODO: set server with tls
+		//filepath.Join(testCertDir, "metron.crt"),
+		//filepath.Join(testCertDir, "metron.key"),
+		//filepath.Join(testCertDir, "loggregator-ca.crt"),
+		//
```

Comment:
TODO?

Reply:
In this PR we are using the loggregator library that already tests tls connection, even though establishing the syslog via tls in the unit test would be closer to the real implementation, we test this through acceptance tests passing and the broker configuring syslog via tls. 
I remove this TODO from this PR in 8338051a0 as I am already testing the creation of a syslog-tls Emitter vs a syslog only emitter.